[Zheng Shijie] iP

SJZheng-Abel opened the pull request.

zhengshijie committed:
level 0

zhengshijie committed:
level 1

zhengshijie committed:
level 2

zhengshijie committed:
level 2

zhengshijie committed:
level 3

zhengshijie committed:
level 4

zhengshijie committed:
A-TextUiTesting

zhengshijie committed:
A-CodingStandard

zhengshijie committed:
Level-5

zhengshijie committed:
A-CodeQuality

zhengshijie committed:
A-Packages

skyventus commented on src/main/java/Duke.java:
```diff
@@ -19,16 +19,16 @@ public static void main(String[] args) {
         ArrayList<Task> tasks = new ArrayList<>();
 
-        while(true){
+        while (true) {
```
If there's a condition to exit the program, would it be better to place the condition inside here? In terms of readability. 
while(! isByeCommand). 
Not saying is wrong, but it would beneficial to people reading the code.

skyventus commented on src/main/java/Duke.java:
```diff
@@ -41,45 +41,42 @@ public static void main(String[] args) {
                 }
-            }catch(Exception e){
+            } catch (Exception e) {
                 System.out.println("Internal error occurs,please re-enter the command.");
```
By catching this exception, will you be able to find out what's wrong or which function is causing the error? 
And should it continue when it hit an error?

skyventus reviewed (commented):
Good Job! doesn't seem to have "code" violations. However, I would suggest leaving some comment in codes would help your colleague understand faster while going through the code.

zhengshijie committed:
branch-A-CodeQuality

zhengshijie committed:
testing

zhengshijie committed:
testing branch

zhengshijie committed:
testing branch

zhengshijie committed:
testing branch

zhengshijie committed:
Merge branch 'testBranch'

# Conflicts:
#	src/main/java/Duke/Tasks/ToDos.java

zhengshijie committed:
restore from testing branch

zhengshijie committed:
Level 6

SJZheng-Abel committed:
Merge pull request #1 from ZhengShijieNUS/branch-Level-6

Level 6

Chilaiping commented on src/main/java/Duke.java:
```diff
@@ -39,7 +39,7 @@ public static void main(String[] args) {
                         break;
                     case "done":
```
Maybe we should align switch statement and Case statement as suggested by Java Coding Standard?

zhengshijie committed:
Level 7

zhengshijie committed:
A-JavaDoc

zhengshijie committed:
Level-7

zhengshijie committed:
branch-Level-7-JavaDoc

# Conflicts:
#	src/main/java/Duke.java

zhengshijie committed:
add ignore list

add ignore list

zhengshijie committed:
A-MoreOOP

zhengshijie committed:
Level-8

SJZheng-Abel committed:
Merge pull request #2 from ZhengShijieNUS/branch-Level-8

Level-8

zhengshijie committed:
Level-9

SJZheng-Abel committed:
Merge pull request #3 from ZhengShijieNUS/branch-Level-9

Level-9

zhengshijie committed:
A-JUnit & fixed bug in Parse

SJZheng-Abel committed:
Merge pull request #4 from ZhengShijieNUS/branch-A-JUnit

A-JUnit

SJZheng-Abel committed:
Set theme jekyll-theme-cayman

SJZheng-Abel committed:
Update README.md

Commit 1st version of the doc

SJZheng-Abel committed:
Update README.md

Make the doc to be a nicer look

zhengshijie committed:
add gradle support

zhengshijie committed:
code style upgrade

modify code style to comply with standard style

zhengshijie committed:
rename test folder

zhengshijie committed:
Update settings.gradle

zhengshijie committed:
GUI 1st version

zhengshijie committed:
GUI version 2

function migration completed

zhengshijie committed:
Update build.gradle

zhengshijie committed:
Debug

fixed no file loaded issue

zhengshijie committed:
Create gradle.yml

zhengshijie committed:
Update Duke.java

Add main method to Duke.java for passing in CI

zhengshijie committed:
Update gradle.yml

to remove unnecessary check part

zhengshijie committed:
Update gradle.yml

remove run part only

zhengshijie committed:
Update gradle.yml

zhengshijie committed:
Update build.gradle

change source compatibility

zhengshijie committed:
Add extension tag

1. add extension tag
2. fixed dialog couldn't display long string bugs

zhengshijie committed:
Make gradlew executable

zhengshijie committed:
Refactor tag file name

zhengshijie committed:
refactor file

zhengshijie committed:
Refactor Duke and Parser

Shorten methods in Duke

zhengshijie committed:
Update Parser.java

rectify checkstyle issue

zhengshijie committed:
Refine tag function

Refine tag function, To make sure tag could be stored and load from file also.

zhengshijie committed:
Rectify bugs

zhengshijie committed:
Refine GUI

Refine GUI:
add background pic,
adjust font size
adjust button background

zhengshijie committed:
Resolve compatibility issue

1. resolve compatibility issue on font
2. optimize logic, enter bye will exit the program directly

zhengshijie committed:
rectify checkstyle issue

SJZheng-Abel committed:
Update and rename README.md to User Guide.md

Add GUI description
Add tag functionality

SJZheng-Abel committed:
Update User Guide.md

refine user guide

SJZheng-Abel committed:
Update User Guide.md

refine user guide

SJZheng-Abel committed:
Create README.md

Create README landing page

SJZheng-Abel committed:
Rename User Guide.md to UserGuide.md

SJZheng-Abel committed:
Create DeveloperGuide.md

1st draft of Developer Guide

zhengshijie committed:
Create Ui.png

SJZheng-Abel committed:
Update DeveloperGuide.md

zhengshijie committed:
Update developer guide

add architecture diagram and update developer guide

zhengshijie committed:
Update architectureDiagram.png
